x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -208,27 +208,31 @@ public void testCountFieldWithFilter() {
     }
 
     /**
-     * Expects - for now
+     * Expects
      * LimitExec[500[INTEGER]]
-     * \_AggregateExec[[],[COUNT(hidden_s{r}#8) AS c],FINAL,null]
+     * \_AggregateExec[[],[COUNT(salary{f}#20) AS c],FINAL,null]
      *   \_ExchangeExec[[count{r}#25, seen{r}#26],true]
-     *     \_AggregateExec[[],[COUNT(hidden_s{r}#8) AS c],PARTIAL,8]
-     *       \_EvalExec[[salary{f}#20 AS s, s{r}#3 AS hidden_s]]
-     *         \_FieldExtractExec[salary{f}#20]
-     *           \_EsQueryExec[test], query[{"esql_single_value":{"field":"emp_no","next":{"range":{"emp_no":{"lt":10050,"boost":1.0}}}}}]
-     *             [_doc{f}#42], limit[], sort[] estimatedRowSize[16]
+     *     \_EsStatsQueryExec[test], stats[Stat[name=salary, type=COUNT, query={
```

Comment:
This is a great example on how this PR helps - since the eval goes away, there's no field extraction needed.